File: test_obs.py
import sys
import subprocess

try:
    import obsws_python as obs
except ImportError:
    print("Installing obsws-python...")
    subprocess.check_call([sys.executable, "-m", "pip", "install", "obsws-python"])
    import obsws_python as obs

try:
    # Connect to OBS WebSocket
    cl = obs.ReqClient(host='192.168.1.3', port=4455, password='L8pHKDHpASEbBCUP')
    
    # Get basic info
    resp = cl.get_version()
    print(f"SUCCESS! Connected to OBS version: {resp.obs_version}")
    
    # Get scenes
    scenes = cl.get_scene_list()
    print(f"Current active scene: {scenes.current_program_scene_name}")
    print("\nAll available scenes:")
    for scene in scenes.scenes:
        print(f" - {scene['sceneName']}")
        
except Exception as e:
    print(f"Failed to connect: {e}")

File: test_obs.mjs
import OBSWebSocket from 'obs-websocket-js';

const obs = new OBSWebSocket();

async function connect() {
  try {
    console.log("Attempting to connect to OBS...");
    const {
      obsWebSocketVersion,
      negotiatedRpcVersion
    } = await obs.connect('ws://192.168.1.3:4455', 'NgyZtWX8IDWIznDe');
    
    console.log(`\x1b[32mSUCCESS!\x1b[0m Connected to OBS WebSocket (v${obsWebSocketVersion})`);
    
    // Get list of scenes
    const sceneList = await obs.call('GetSceneList');
    console.log(`\nCurrent active scene: \x1b[36m${sceneList.currentProgramSceneName}\x1b[0m`);
    console.log("All scenes:");
    sceneList.scenes.forEach(scene => {
      console.log(` - ${scene.sceneName}`);
    });

    await obs.disconnect();
  } catch (error) {
    console.error('\x1b[31mFailed to connect\x1b[0m:', error.code, error.message);
  }
}

connect();
